microsoft-edge/webview2/how-to/print.md

```diff
@@ -0,0 +1,744 @@
+  CHECK_FAILURE(printSettings2->put_PageRanges(samplePrintSettings.Pages.c_str()));
+  if (samplePrintSettings.Media == COREWEBVIEW2_PRINT_MEDIA_SIZE_CUSTOM)
+  {
+    CHECK_FAILURE(printSettings2->put_PageWidth(samplePrintSettings.PaperWidth));
+    CHECK_FAILURE(printSettings2->put_PageHeight(samplePrintSettings.PaperHeight));
+  }
+  CHECK_FAILURE(printSettings2->put_ColorMode(samplePrintSettings.ColorMode));
+  CHECK_FAILURE(printSettings2->put_Collation(samplePrintSettings.Collation));
+  CHECK_FAILURE(printSettings2->put_Duplex(samplePrintSettings.Duplex));
+  CHECK_FAILURE(printSettings2->put_ScaleFactor(samplePrintSettings.ScaleFactor));
+  CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(samplePrintSettings.PrintBackgrounds));
+  CHECK_FAILURE(printSettings2->put_ShouldPrintHeaderAndFooter(samplePrintSettings.HeaderAndFooter));
+  CHECK_FAILURE(printSettings2->put_HeaderTitle(samplePrintSettings.HeaderTitle.c_str()));
+  CHECK_FAILURE(printSettings2->put_FooterUri(samplePrintSettings.FooterUri.c_str()));
+  CHECK_FAILURE(printSettings2->put_PrinterName(printerName.c_str()));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  CHECK_FAILURE(webView2_16->Print(
+      printSettings.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = std::move(title), this](HRESULT errorCode,
+                                   COREWEBVIEW2_PRINT_STATUS printStatus) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && printStatus == COREWEBVIEW2_PRINT_STATUS_SUCCEEDED)
+            {
+              message = L"Printing " + std::wstring(title.get()) +
+                              L" document to printer is succedded";
+            }
+            else if (errorCode == S_OK && printStatus == COREWEBVIEW2_PRINT_STATUS_PRINTER_UNAVAILABLE)
+            {
+              message = L"Selected printer is not found, not available, offline or "
+                              L"error state.";
+            }
+            else if (errorCode == E_INVALIDARG)
+            {
+              message = L"Invalid settings provided for the specified printer";
+            }
+            else if (errorCode == E_ABORT)
+            {
+              message = L"Printing " + std::wstring(title.get()) +
+                                  L" document already in progress";
+            }
+            else
+            {
+              message = L"Printing " + std::wstring(title.get()) +
+                                  L" document to printer is failed";
+            }
+
+            AsyncMessageBox(message, L"Print to printer");
+
+            return S_OK;
+          })
+          .Get()));
+  return true;
+}
+```
+
+
+---
+
+<!-- end of tab-set -->
+
+
+<!-- ====================================================================== -->
+## The PrintToPdf method to customize printing to a PDF file
```

Comment:
Can you provide more clarity on what we are customizing? Print settings, print dialog, etc?